onnxruntime/hosting/server_configuration.h

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef ONNXRUNTIME_HOSTING_SERVER_CONFIGURATION_H
+#define ONNXRUNTIME_HOSTING_SERVER_CONFIGURATION_H
+
+#include <thread>
+#include <fstream>
+#include <unordered_map>
+
+#include "boost/program_options.hpp"
+#include "core/common/logging/logging.h"
+
+namespace onnxruntime {
+namespace hosting {
+
+namespace po = boost::program_options;
+
+enum class Result {
+  ExitSuccess = 1,
+  ExitFailure,
+  ContinueSuccess
+};
+
+static std::unordered_map<std::string, onnxruntime::logging::Severity> supported_log_levels{
+    {"verbose", onnxruntime::logging::Severity::kVERBOSE},
+    {"info", onnxruntime::logging::Severity::kINFO},
+    {"warning", onnxruntime::logging::Severity::kWARNING},
+    {"error", onnxruntime::logging::Severity::kERROR},
+    {"fatal", onnxruntime::logging::Severity::kFATAL}};
+
+// Wrapper around Boost program_options and should provide all the functionality for options parsing
+// Provides sane default values
+class ServerConfiguration {
+ public:
+  const std::string full_desc = "ONNX Hosting: host an ONNX model for inferencing with ONNXRuntime";
+  std::string model_path;
+  std::string address = "0.0.0.0";
+  int http_port = 8001;
+  int num_http_threads = std::thread::hardware_concurrency();
+  onnxruntime::logging::Severity logging_level{};
+
+  ServerConfiguration() {
+    desc.add_options()("help,h", "Shows a help message and exits");
+    desc.add_options()("logging_level", po::value(&logging_level_str)->default_value(logging_level_str), "Logging level. Allowed options (case sensitive): verbose, info, warning, error, fatal");
+    desc.add_options()("model_path,m", po::value(&model_path)->required(), "Path to ONNX model");
+    desc.add_options()("address,a", po::value(&address)->default_value(address), "The base HTTP address");
+    desc.add_options()("http_port", po::value(&http_port)->default_value(http_port), "HTTP port to listen to requests");
```

Comment:
`,p`?

Reply:
I didn't add `-p` as a program arg because it'll be confusing when we have `grpc_port`. I think we'll just move forward with no short args for now. Removing all of them `-h`